### DISCUSSION TITLE
Retrieving encrypted metadata backups

### DISCUSSION THREAD

**Discussion by bbonvi** [juicedata/juicefs#3141] [Q&A]
Hello.
I assume if the file system volume uses an encryption, so do the metadata auto backups under the `/meta` folder. Is this correct? In this case, how do i decrypt them back? I can't use `gzip -d` on `.json.gz` file since it gives me `Input/output error` and of course i can't use `juicefs load`.

**Comment by greemo** [comment]
Hi All, I'm also trying to figure out how to decrypt the metadata dumps. This is a hurdle to be able to use juicefs, as I need to test backup retrieval.

Looking at the code, it seems to just call object_store.Put, which in the case of an encrypted object store, means the key to decrypt is put in the database, meaning the decryption is impossible??? Hopefully I understand this incorrectly.

Could a symmetrical encryption of these dump files with the encryption key be performed? Then the user just needs to backup the encryption key, and everything else is taken care of.

**Reply by davies** [reply]
We can't do that easily even you have the encryption key, will fix it soon.

**Comment by zhijian-pro** [comment]
close by #3311
